pkg/printers/customcolumn_flags.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+)
+
+// CustomColumnsPrintFlags provides default flags necessary for printing
+// custom resource columns from an inline-template or file.
+type CustomColumnsPrintFlags struct {
+	NoHeaders        *bool
+	TemplateArgument *string
+}
+
+// ToPrinter receives an outputFormat and returns a printer capable of
+// handling custom-column printing.
+// Returns false if the specified outputFormat does not match a supported format.
+// Supported format types can be found in pkg/printers/printers.go
+func (f *CustomColumnsPrintFlags) ToPrinter(outputFormat string) (ResourcePrinter, bool, error) {
+	if len(outputFormat) == 0 {
+		return nil, false, fmt.Errorf("missing output format")
+	}
+
+	templateArg := ""
+
+	supportedFormats := map[string]bool{
+		"custom-columns-file": true,
+		"custom-columns":      true,
+	}
+
+	if f.TemplateArgument == nil || len(*f.TemplateArgument) == 0 {
+		for format := range supportedFormats {
+			format = format + "="
+			if strings.HasPrefix(outputFormat, format) {
+				templateArg = outputFormat[len(format):]
```

Comment:
Please use two distinct name, similarly like in #61304 so it's clear which is template format and which its value, aside from `outputFormat` which is your input parameter.